openupgrade_scripts/scripts/account/17.0.1.2/end-migration.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2025 Le Filament (https://le-filament.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def _assign_journal_xmlids(env):
+    """
+    Starting in Odoo 17, Odoo expects journals to get external id
+    like account.1_sale for sale journal on company 1
+    We therefore assign xmlids to existing journals so that they will not be recreated
+    """
+    for company in env["res.company"].search([("chart_template", "!=", False)]):
+        template_data = env["account.chart.template"]._get_chart_template_data(
+            company.chart_template
+        )
+        existing_journals = (
+            env["account.journal"]
+            .with_context(active_test=False)
+            .search(env["account.journal"]._check_company_domain(company), order="id")
+        )
+        for xmlid, journal_data in list(
+            template_data.get("account.journal", {}).items()
+        ):
+            if not env.ref(xmlid, raise_if_not_found=False) and "type" in journal_data:
+                journal = (
+                    existing_journals.filtered(lambda j: j.type == journal_data["type"])  # noqa: B023
+                    or [False]
+                )[0]
+                if journal:
```

Comment:
```suggestion
                    existing_journals.filtered(lambda j: j.type == journal_data["type"])  # noqa: B023
                )
                if journal:
```
empty record already is not True

Reply:
Thanks @MiquelRForgeFlow 
My problem here is that I only want the first record (when the filtered() may return a recordset of multiple records)
So I added a [0], but if filtered() returns empty recordset, the [0] is returning an error, which is why I replaced empty recordset by [False] !
There is probably a more elegant way to do that that I do not know !

Reply:
`[:1]` does the trick.